File: bindings/node/examples/remote_query.mjs
import {
  CogentClient,
  addGatewayRemote,
  printTextRun,
  readRemoteArgs,
  textOptions,
} from './_common.mjs';

const { alias, input } = readRemoteArgs('Write one sentence about remote inference.');
const client = new CogentClient();
const endpoint = addGatewayRemote(client, alias);
await printTextRun(client.query({
  endpoint,
  prompt: input,
  options: textOptions(),
}));

File: bindings/python/examples/remote_query.py
from __future__ import annotations

from _common import (
    CogentClient,
    add_gateway_remote,
    print_text,
    read_remote_args,
    text_options,
)


def main() -> None:
    alias, prompt = read_remote_args("Write one sentence about remote inference.")
    client = CogentClient()
    endpoint = add_gateway_remote(client, alias)
    run = client.query(
        prompt,
        endpoint=endpoint,
        options=text_options(),
    )
    print_text(run.result())


if __name__ == "__main__":
    main()
